lib/Dialect/TTNN/IR/TTNNOps.cpp

```diff
@@ -107,6 +107,57 @@ ::mlir::LogicalResult mlir::tt::ttnn::TransposeOp::verify() {
   return success();
 }
 
+::mlir::LogicalResult mlir::tt::ttnn::ReshapeOp::verify() {
+  ::mlir::RankedTensorType inputType = getInput().getType();
+  ::mlir::RankedTensorType outputType = getOutput().getType();
+  ArrayAttr shape = getShape();
+  int64_t shape_size = static_cast<int64_t>(shape.size());
+
+  if (shape_size == 0) {
+    return emitOpError("Shape attribute must be non-empty");
+  }
+
+  if (shape_size > 5) {
+    return emitOpError("Shape attribute must have at most 5 elements");
+  }
+
+  if (inputType.getNumElements() != outputType.getNumElements()) {
+    return emitOpError(
+        "Input and output tensors must have the same number of elements");
+  }
+
+  bool has_negative = false;
+  for (int64_t i = 0; i < shape_size; i++) {
+    if (mlir::cast<IntegerAttr>(shape[i]).getInt() == -1) {
+      if (has_negative) {
+        return emitOpError("Shape attribute must have at most one -1 element");
+      }
+
+      has_negative = true;
+    }
+  }
+
+  // If there is no -1 in the shape, check that the shape matches the output
+  // shape.
```

Comment:
I'm probably missing something - shouldn't the shape match the output shape regardless of whether there's a -1?